### PR TITLE
Set Multiple Options in one pass, and at NEW.

### PR DESCRIPTION
**Pull request by thebeline**
Ability to set multiple Options at once and at instantiation.

### CODE REVIEW

File: src/Curl/Curl.php [hunk 1]
```diff
@@ -105,9 +105,11 @@ public function __construct($base_url = null)
...
         $this->setDefaultJsonDecoder();
         $this->setDefaultTimeout();
-        $this->setOpt(CURLINFO_HEADER_OUT, true);
-        $this->setOpt(CURLOPT_HEADERFUNCTION, array($this, 'headerCallback'));
-        $this->setOpt(CURLOPT_RETURNTRANSFER, true);
+        $this->setOpts($options + array(
+            CURLINFO_HEADER_OUT => true,
+            CURLOPT_HEADERFUNCTION => array($this, 'headerCallback'),
+            CURLOPT_RETURNTRANSFER => true
+        ));
```
Comment (zachborboa) [inline src/Curl/Curl.php]: I believe we should set default options first. Then apply the user's preferred options after.

```php
        $this->setOpt(CURLINFO_HEADER_OUT, true);
        $this->setOpt(CURLOPT_HEADERFUNCTION, array($this, 'headerCallback'));
        $this->setOpt(CURLOPT_RETURNTRANSFER, true);
        $this->setOpts($options);
```

File: src/Curl/Curl.php [hunk 2]
```diff
@@ -765,7 +754,57 @@ public function setOpt($option, $value)
         }
 
         $this->options[$option] = $value;
-        return curl_setopt($this->curl, $option, $value);
+        return (bool) curl_setopt($this->curl, $option, $value);
```
Comment (zachborboa) [inline src/Curl/Curl.php]: Why is `(bool)`added here?

File: src/Curl/Curl.php (hunk 1 above)
Comment (thebeline) [inline src/Curl/Curl.php]: So, swap the params around `+`?

File: src/Curl/Curl.php (hunk 2 above)
Comment (thebeline) [inline src/Curl/Curl.php]: (sometimes) I try to force ably cast return values if I can.  It's, um, a weird habit.  Follows along with my too many spaces in docblocks.

**Comment by thebeline** [comment]
@zachborboa - I am going to swap those params, beyond that, is there something blocking this? (I am asking this on a few)

**Comment by thebeline** [comment]
@zachborboa - ack, that's what happens when one walks away for a bit.  Your concern about the `+` operator is exactly as I expected, the user values override the defaults (values preceding the `+` do not get overwritten).

https://secure.php.net/manual/en/language.operators.array.php

That said, I can see there being a possible issue if the user's preferred options are incorrect/fail, it may leave the object in an undesirable state.  Should we throw an error instead at that point?  If there is an error?  Currently, even if we set the defaults first, then the user's second, there will be no warning.  I would almost rather:

```
if(!$this->setOpts($opts + $default_opts)) {
    throw new \Exception();
}
```

What do you think?

**Comment by zachborboa** [comment]
`getOpt`and `setOpts` have been implemented. Thanks for the pull request.

**Pull request closed without merging**